### PR TITLE
Move metrics creation

### PR DESCRIPTION
**Pull request by pacospace**
Signed-off-by: Francesco Murdaca <fmurdaca@redhat.com>

## Related Issues and Dependencies

Fixes: https://github.com/thoth-station/user-api/issues/1367
Fixes: https://github.com/thoth-station/user-api/issues/1345
Related-To: https://github.com/thoth-station/thoth-application/issues/1112

## This introduces a breaking change

- [ ] Yes
- [x] No

### CODE REVIEW

File: thoth/user_api/api_v1.py [hunk 1]
```diff
@@ -43,7 +43,10 @@
 from thoth.python import Project
 from thoth.python.exceptions import ThothPythonException
 from thoth.user_api.payload_filter import PayloadProcess
-from thoth.user_api.openapi_server import metrics
+from thoth.user_api.openapi_server import _METRIC_CACHE_HIT_ADVISER_AUTHENTICATED
```
Comment (fridex) [inline thoth/user_api/api_v1.py:46]: The module from which the import is done is still the same. Will this address the issue?

Comment (pacospace) [reply thoth/user_api/api_v1.py:46]: correct, then prometheus_flask_exporter expects metrics to be present all in openapi_server, we cannot import it and use it in a method, we should do something like

the best solution using `prometheus_flask_exporter` seems:

@app.route('/advise/python')
@metrics.counter(
    'thoth_user_api_cache_hit_rate',
    'Thoth User API cache hit rate',
    labels={
        'is_authenticated': lambda: response['authenticated'],
        'service': "adviser",
        `env`: Configuration.THOTH_DEPLOYMENT_NAME
    })
def advise_python(inputs):
    return advise_python

but it does not seem the right way.

Comment (pacospace) [reply thoth/user_api/api_v1.py]: also `from .openapi_server import metrics` failed @fridex :/

Comment (pacospace) [reply thoth/user_api/api_v1.py]: maybe I have another idea

Comment (pacospace) [reply thoth/user_api/api_v1.py]: Using after_request, wdyt @fridex ?

**Comment by goern** [comment]
is this WIP?

**Review by fridex — APPROVED** [review]
nice, lgtm 👍🏻

**Comment by pacospace** [comment]
> is this WIP?

could not be tested, because the cluster is down, but in principle it was ready.

**Pull request merged**